src/vs/workbench/contrib/terminal/browser/terminalActions.ts

```diff
@@ -491,7 +491,11 @@ export function registerTerminalActions() {
 				return;
 			}
 			c.service.setActiveInstance(instance);
-			return c.groupService.showPanel(true);
+			if (instance.target === TerminalLocation.Panel) {
+				c.groupService.showPanel(true);
+			} else {
+				c.editorService.activeInstance?.focus();
+			}
```

Comment:
There's a helper for this:

https://github.com/microsoft/vscode/blob/a40ee0e2fa08d749d39cffab1824a29efbf27e66/src/vs/workbench/contrib/terminal/browser/terminalActions.ts#L1835-L1842